### PR TITLE
Rework CI

### PR DESCRIPTION
**Pull request by djphoenix**
- Add more compilers - especially GCC on macOS and Clang.
- Add more architectures - especially RISC-V-64
- Move build outside of QEMU (use cross-compilation)
- Use qemu-user for test on target environment
- Add Android CI with APK artifacts

### CODE REVIEW

**Comment by djphoenix** [comment]
Oh... on my fork, ubuntu-latest was a jammy, and here I see a focal. And no gcc-12 here. What..?

**Comment by djphoenix** [comment]
aarch64 with thread sanitizer takes too long... Should we keep it here?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,262 +7,226 @@ on:
...
+            compiler: clang
+            sanitizer: leak # Not supported by Apple Clang
+          - os: macos
+            compiler: clang
+            sanitizer: memory # Not supported by Apple Clang
+          - os: macos
+            compiler: gcc
+            sanitizer: thread # Failing on CI, should be supported?
+
+          # Vangrind is unavailable for cross-builds
```
Comment (alex19EP) [inline .github/workflows/ci.yml:112]: should it be:
```suggestion
          # Valgrind is unavailable for cross-builds
```

File: .github/workflows/dist.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+name: Distributes
```
Comment (alex19EP) [inline .github/workflows/dist.yml:1]: maybe shorten it to:

```suggestion
name: Dist
```

**Review by alex19EP — CHANGES_REQUESTED** [review]
some minor nitpicking from me.

**Review by alex19EP — APPROVED** [review]
LGTM

**Comment by djphoenix** [comment]
@jaacoppi @sthibaul any comments?

**Comment by sthibaul** [comment]
LGTM as well, thanks!

**Pull request merged**